src/main/routes/logout.ts

```diff
@@ -2,13 +2,30 @@ import * as express from 'express'
 import * as config from 'config'
 import * as Cookies from 'cookies'
 
+import { IdamClient } from 'app/idam/idamClient'
+import { Logger } from '@hmcts/nodejs-logging'
+
 import { Paths } from 'app/paths'
+import { JwtExtractor } from 'idam/jwtExtractor'
+import { JwtUtils } from 'common/utils/jwtUtils'
 
 const sessionCookie = config.get<string>('session.cookieName')
+const logger = Logger.getLogger('routes/logout')
 
 /* tslint:disable:no-default-export */
 export default express.Router()
-  .get(Paths.logoutReceiver.uri, (req: express.Request, res: express.Response, next: express.NextFunction) => {
+  .get(Paths.logoutReceiver.uri, async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
+    const jwt: string = JwtExtractor.extract(req)
+
+    if (jwt) {
+      try {
+        await IdamClient.invalidateCode(jwt)
+      } catch (error) {
+        const { id } = JwtUtils.decodePayload(this.bearerToken)
```

Comment:
Method can throw the following error, please catch it by wrapping route handled with `ErrorHandling.apply` function. Of course that will mean that no cookie will get deleted when.decoding failed but I am OK with that since this is very rare case and probably session has been manually set to something that is not valid session / JWT anyway.

 ```
throw new Error(`Unable to parse JWT token: ${jwt}`)
```

Reply:
done